Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/Parse/Inlines/EmojiInline.cs

```diff
@@ -0,0 +1,922 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System.Collections.Generic;
+using Microsoft.Toolkit.Uwp.UI.Controls.Markdown.Helpers;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls.Markdown.Parse
+{
+    internal class EmojiInline : MarkdownInline, IInlineLeaf
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmojiInline"/> class.
+        /// </summary>
+        public EmojiInline()
+            : base(MarkdownInlineType.Emoji)
+        {
+        }
+
+        // Codes taken from https://gist.github.com/rxaviers/7360908
+        // Ignoring not implented symbols in Segoe UI Emoji font (e.g. :bowtie:)
+        private static readonly Dictionary<string, int> _emojiCodesDictionary = new Dictionary<string, int>
```

Comment:
Is it ok to have this dictionary here or should I make this class partial and move dictionary to another file?

Reply:
I was actually wondering how we should manage it.  Splitting it out at least into a separate partial class is a good idea.

Reply:
why do you need an _emojiCodesDictionary ?

Reply:
@hermitdave it's what translates `:smile:` into the actual character needed by the font for display :smile:.

Reply:
Oh crap.. GitHub didn't show me that file at all.

Reply:
@hermitdave it's mapping between textual representation of emoji (e.g. : smile :) and Unicode char from Segoe UI Emoji font